File: hash/go/imagehash/internal/testkit/specpath.go
package testkit

import (
	"path/filepath"
)

// SpecDir resolves /spec/ relative to the calling test's working directory.
// Go tests run with CWD set to the package directory containing the test file,
// so internal/* tests need "../../../spec" and root-package tests need "../../spec".

// DirRoot returns the spec dir as resolved from the root-package CWD (go/imagehash/).
func DirRoot() string { return filepath.Join("..", "..", "spec") }

// DirInternal returns the spec dir as resolved from an internal/* package CWD.
func DirInternal() string { return filepath.Join("..", "..", "..", "..", "spec") }

// PathFromRoot joins SpecDir + relative.
func PathFromRoot(rel string) string { return filepath.Join(DirRoot(), rel) }

// PathFromInternal joins SpecDir + relative for internal/* tests.
func PathFromInternal(rel string) string { return filepath.Join(DirInternal(), rel) }

File: hash/rust/rosetta-squint-hash/tests/testkit/spec_path.rs
//! Relative path constants. Cargo runs integration tests with CWD = crate root,
//! so /spec/ is resolved via "../../spec".

pub const SPEC_DIR: &str = "../../spec";

pub fn spec_path(rel: &str) -> String {
    format!("{}/{}", SPEC_DIR, rel)
}
